src/josepy/util.py

```diff
@@ -151,6 +151,9 @@ def __hash__(self) -> int:
         raise NotImplementedError()
 
 
+T = TypeVar('T', bound='ImmutableMap')
```

Comment:
I think/hope no one would be confused by this, but what do you think about prefixing this variable with an underscore to make it clear it's not part of the module's API.

Reply:
See above.

Reply:
Done with `GenericImmutableMap`.